### PR TITLE
Implementation of VField.setValueID()

### PR DESCRIPTION
**Pull request by hfazai**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 1]
```diff
@@ -2041,36 +2043,42 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
-              }
-            }
-          }
-          break
-        } catch (e: SQLException) {
-        } catch (error: Error) {
-        } catch (rte: RuntimeException) {
-        }
+    val result = try {
+      transaction {
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:2054]: remove transaction

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 2]
```diff
@@ -2041,36 +2043,42 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
+        val table= evalListTable()
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:2055]: add space after table

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 3]
```diff
@@ -2041,36 +2043,42 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
+        val idColumn = table.columns.find { it.name == "ID" } as Column<Int>
+        val firstRecord = table.slice(table.resolveColumn(list!!.getColumn(0).column!!)).select {
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:2057]: if VListColumn contain an val cloumn_ de type exposed.column (sarrah pull request) so you don't need to fetch column here !

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithListDomainsTests.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  @Test
+  fun setValueIDTest() {
+    FormWithListDomains.model
+    val model = FormWithListDomains.userListBlock.user.vField
+
+    model.setValueID(1)
+    assertEquals(1, FormWithListDomains.userListBlock.user.value)
+
+    model.setValueID(0)
+    assertEquals(null, FormWithListDomains.userListBlock.user.value)
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/FormWithListDomainsTests.kt:52]: open transaction here

**Review by h-haddad — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt (hunk 1 above)
Comment (hfazai) [reply galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:2054]: it's `startProtected`

**Review by h-haddad — APPROVED** [review]

**Pull request merged**